tests/tests/neg/surface/vec02.rs

```diff
@@ -2,9 +2,18 @@
 
 use std::ops::Index;
 
+#[path = "../../lib/option.rs"]
+mod option;
+
+#[path = "../../lib/slice.rs"]
+mod slice;
+
```

Comment:
Are these extra imports needed?

I hate the tests that import other files, because if something fails on those files, a bunch of tests fail, and it's hard to see at a glance what's wrong. We should keep tests as self-contained as possible.

Reply:
The `option` is needed for the `Iterator` which uses the indexed-option.

I've split the test up into those for `iter_slice` and `iter_vec` where each only requires the corresponding import.